### PR TITLE
OEL-1651: Published date prefilled with current date. 

### PR DESCRIPTION
**Pull request by escuriola**
https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-1651

### CODE REVIEW

File: tests/src/Functional/NewsIntegrationTest.php [hunk 1]
```diff
@@ -91,6 +93,16 @@ public function testCreateNews() {
     $assert_session->pageTextContains('01/24/2022');
     $assert_session->responseContains('Starter Image test');
     $assert_session->responseContains('Starter Image caption');
+
+    // Assert publication date is current date if it's not fulfilled.
+    $this->drupalGet('node/1/edit');
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php:98]: We rely on the node id being 1?
Seems safe, so let's keep it.

**Review by brummbar — CHANGES_REQUESTED** [review]
image

File: tests/src/Functional/NewsIntegrationTest.php [hunk 2]
```diff
@@ -91,6 +93,16 @@ public function testCreateNews() {
...
+    $page->fillField('Date', '');
+    $page->pressButton('Save');
+
+    $assert_session->pageTextContains(DrupalDateTime::createFromTimestamp(
+      \Drupal::time()->getRequestTime(),
+      DateTimeItemInterface::STORAGE_TIMEZONE
+    )->format('D, m/d/Y'));
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php]: This seems a bit fragile.
If the display timezone is different from the storage timezone, or if the test runs at midnight, then we might get an unexpected result here.

In the planning I remember we said we should only test that there is any date, not which date specifically.

And i think the better way to test this is to first create the node without setting a pub date, and then save it again with a specific pub date.

File: modules/oe_starter_content_news/oe_starter_content_news.module [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return;
+  }
+
+  $entity->set(
+    'oe_publication_date',
+    DrupalDateTime::createFromTimestamp(
+      \Drupal::time()->getRequestTime(),
+      DateTimeItemInterface::STORAGE_TIMEZONE
+    )->format(DateTimeItemInterface::DATE_STORAGE_FORMAT),
+  );
```
Comment (donquixote) [inline modules/oe_starter_content_news/oe_starter_content_news.module]: Seems we both misremember the planning / refinement.
I talked with Francesco, we should use default value through the field configuration - which makes sense.
we then need an upgrade path to update the field config with the default value.

File: tests/src/Functional/NewsIntegrationTest.php (hunk 2 above)
Comment (donquixote) [reply tests/src/Functional/NewsIntegrationTest.php]: We could use https://www.drupal.org/project/datetime_testing but seems too much work for this issue.

File: tests/src/Functional/NewsIntegrationTest.php [hunk 4]
```diff
@@ -73,24 +76,33 @@ public function testCreateNews() {
...
+    // is current date because it's not fulfilled.
     $assert_session->pageTextContains('News Example title has been created.');
     $assert_session->pageTextContains('Example title');
     $assert_session->pageTextContains('Example Content');
     $assert_session->pageTextContains('Example Introduction');
-    $assert_session->pageTextContains('01/24/2022');
+    $assert_session->pageTextContains(DrupalDateTime::createFromTimestamp(
+      \Drupal::time()->getRequestTime(),
+      DateTimeItemInterface::STORAGE_TIMEZONE
+    )->format('m/d/Y'));
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php]: I still don't trust this part, it could lead to unexpected fail if the test runs at the wrong time.
Either we test this in a more predictable way, or we test only that the field is not empty.
See also my older comment which is now outdated.

File: tests/src/Functional/NewsIntegrationTest.php [hunk 5]
```diff
@@ -73,24 +76,33 @@ public function testCreateNews() {
...
     $assert_session->responseContains('Starter Image test');
     $assert_session->responseContains('Starter Image caption');
+
+    // Assert publication date when fulfilled.
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php]:
```suggestion
    // Set a custom publication date.
```

File: tests/src/Functional/NewsIntegrationTest.php (hunk 4 above)
Comment (donquixote) [reply tests/src/Functional/NewsIntegrationTest.php]: What you could do:
- select the only `<time>` tag on the page
- use regex to find the `\d+/\d+/\d+` date string in the element text. If you are lazy, you are finished now.
- (don't) If you are more ambitious, you could assert that the date is within 1d range of the current date. This gets rid of possible mistakes. But I think we should not do it, too many moving parts, too much conditional logic.
- (don't) If you are really ambitious you could use https://www.drupal.org/project/datetime_testing and simulate creating a news half hour before midnight and half hour after midnight, or just 1x at noon. we do that in project for the progress bar. But imo we should not spend that extra time + effort.

I leave the last two points for the next reviewer so they can follow the decision process and share their opinion.

File: tests/src/Functional/NewsIntegrationTest.php [hunk 6]
```diff
@@ -73,24 +76,33 @@ public function testCreateNews() {
...
     $page->fillField('Introduction', 'Example Introduction');
-    $page->fillField('Date', '2022-01-24');
     $media_name = $media_image->getName() . ' (' . $media_image->id() . ')';
     $page->fillField('Media item', $media_name);
     $page->fillField('Caption', 'Starter Image caption');
     $page->pressButton('Save');
 
-    // Assert media document has been created.
+    // Assert media document has been created and publication date
+    // is current date because it's not fulfilled.
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php]: "fulfill" does not mean what you think it does.
```suggestion
    // Assert that news was created with the expected field values.
```
Put another comment further below about pub date.

File: tests/src/Functional/NewsIntegrationTest.php [hunk 7]
```diff
@@ -73,24 +76,33 @@ public function testCreateNews() {
...
     $assert_session->pageTextContains('News Example title has been created.');
     $assert_session->pageTextContains('Example title');
     $assert_session->pageTextContains('Example Content');
     $assert_session->pageTextContains('Example Introduction');
```
Comment (donquixote) [inline tests/src/Functional/NewsIntegrationTest.php:89]: We can insert the comment about pub date here - but depends on the solution we choose for testing it.
```suggestion
    $assert_session->pageTextContains('Example Introduction');
    // Assert that publication date was filled with a default value.
```

File: tests/src/Functional/NewsIntegrationTest.php (hunk 6 above)
Comment (donquixote) [reply tests/src/Functional/NewsIntegrationTest.php]: (the old comment was wrong, it talked about media document - or did I miss something?)

File: tests/src/Functional/NewsIntegrationTest.php (hunk 4 above)
Comment (donquixote) [reply tests/src/Functional/NewsIntegrationTest.php]: Btw, if we would use 'classy' theme instead of 'stark', we could select the field using a class.
But it is not really needed here, seems like there is only one `<time>` tag.

File: modules/oe_starter_content_news/oe_starter_content_news.post_update.php [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ * @file
+ * OpenEuropa starter content post updates.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\field\Entity\FieldConfig;
+
+/**
+ * Enable OE Webtools Social share, Cookie consent and Page feedback components.
```
Comment (donquixote) [inline modules/oe_starter_content_news/oe_starter_content_news.post_update.php]: What is this comment?

Comment (escuriola) [reply modules/oe_starter_content_news/oe_starter_content_news.post_update.php]: Sorry! Copy&paste one :dancing_men:

**Review by donquixote — APPROVED** [review]

File: tests/src/Functional/NewsIntegrationTest.php [hunk 9]
```diff
@@ -73,24 +74,32 @@ public function testCreateNews() {
...
-    $assert_session->pageTextContains('01/24/2022');
     $assert_session->responseContains('Starter Image test');
     $assert_session->responseContains('Starter Image caption');
+    // Assert that publication date was filled with a default value.
+    $publication_date = $page->find('css', 'time');
+    $publication_date->hasAttribute('datetime');
+    $this->assertMatchesRegularExpression("/\d+\/\d+\/\d+/", $publication_date->getText());
```
Comment (drishu) [inline tests/src/Functional/NewsIntegrationTest.php:95]: dont like it but since we are out of time it will have to do

**Review by drishu — APPROVED** [review]

**Review by brummbar — APPROVED** [review]
